sycl/include/CL/sycl/intel/sub_group.hpp

```diff
@@ -116,42 +116,38 @@ struct sub_group {
     return __spirv_GroupAll(__spv::Scope::Subgroup, predicate);
   }
 
+
+  template <typename T>
+  using EnableIfIsArithmeticOrHalf = typename std::enable_if<
```

Comment:
The half type is fundamental SYCL type as well as int, float and etc.
So I would suggest to have a specific is_arithmetic for SYCL types.
Lets add into <CL/sycl/detail/type_traits.hpp>

```
template <typename T>
using is_arithmetic = std::integral_constant<bool, std::is_arithmetic<T>::value
    || std::is_same<remove_const_t<T>, half>::value>;
```

And change EnableIfIsArithmeticOrHalf  to 
```
template <typename T>
using EnableIfIsArithmetic = detail::enable_if_t<detail::is_arithmetic<T>::value, T>;
```